### PR TITLE
build(deps): upgrade to Cosmos v0.44.0

### LINKED ISSUE #585: Upgrade to Cosmos v0.43
Recently the Cosmos team has released version [`v0.43.0`](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.43.0). We should upgrade to it.

### PR DESCRIPTION
**Pull request by RiccardoM**
## Description
This PR upgrades Desmos to use Cosmos v0.44.0.
Some tests might need some fixes.

Closes #585

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [ ] Updated the documentation. 
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: app/desmos/cmd/root.go [hunk 1]
```diff
@@ -90,8 +91,67 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
...
+	// validators here.
+	//
+	// In summary:
+	// - if you leave srvCfg.MinGasPrices = "", all validators MUST tweak their
+	//   own app.toml config,
+	// - if you set srvCfg.MinGasPrices non-empty, validators CAN tweak their
+	//   own app.toml to override, or use this default value.
+	//
+	// In simapp, we set the min gas prices to 0.
+	srvCfg.MinGasPrices = "0stake"
```
Comment (RiccardoM) [inline app/desmos/cmd/root.go:129]: This needs to be changed

File: go.mod [hunk 2]
```diff
@@ -3,9 +3,10 @@ module github.com/desmos-labs/desmos
 go 1.15
 
 require (
-	github.com/armon/go-metrics v0.3.8
-	github.com/cosmos/cosmos-sdk v0.42.4
+	github.com/armon/go-metrics v0.3.9
+	github.com/cosmos/cosmos-sdk v0.43.0
```
Comment (RiccardoM) [inline go.mod:7]: Better change this with `v0.44.x`

File: third_party/proto/cosmos/base/query/v1beta1/pagination.proto [hunk 3]
```diff
@@ -30,6 +30,9 @@ message PageRequest {
   // count_total is only respected when offset is used. It is ignored when key
   // is set.
   bool count_total = 4;
+
+  // reverse is set to true if results are to be returned in the descending order.
+  bool reverse = 5;
```
Comment (RiccardoM) [inline third_party/proto/cosmos/base/query/v1beta1/pagination.proto:35]: Didn't see this was added. Nice!

File: x/profiles/simulation/genesis.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ import (
 
 // RandomizedGenState generates a random GenesisState for profile
 func RandomizedGenState(simsState *module.SimulationState) {
-	profilesNumber := simsState.Rand.Intn(len(simsState.Accounts) - 10)
+	profilesNumber := simsState.Rand.Intn(len(simsState.Accounts) - 5)
```
Comment (RiccardoM) [inline x/profiles/simulation/genesis.go:18]: Why was this changed? @dadamu

File: x/profiles/simulation/params.go [hunk 5]
```diff
@@ -18,29 +19,33 @@ func ParamChanges(r *rand.Rand) []simtypes.ParamChange {
...
 				params := RandomOracleParams(r)
+				feeAmountBz, err := json.Marshal(params.FeeAmount)
+				if err != nil {
+					panic(err)
+				}
 				return fmt.Sprintf(
-					`{"script_id":"%d", "ask_count":"%d", "min_count":"%d", "prepare_gas":"%d", "execute_gas":"%d", "fee_payer":"%s", fee_coins:"%s"}`,
-					params.ScriptID, params.AskCount, params.MinCount, params.PrepareGas, params.ExecuteGas, params.FeePayer, params.FeeAmount.String(),
+					`{"script_id":"%d", "ask_count":"%d", "min_count":"%d", "prepare_gas":"%d", "execute_gas":"%d", "fee_payer":"%s", "fee_amount":%s}`,
+					params.ScriptID, params.AskCount, params.MinCount, params.PrepareGas, params.ExecuteGas, params.FeePayer, string(feeAmountBz),
```
Comment (RiccardoM) [inline x/profiles/simulation/params.go:48]: Why was the fee serialization changed @dadamu?

Comment (dadamu) [reply x/profiles/simulation/params.go:48]: Because It should be json format. Proto String() method returns compactText format string instead of JSON string.

File: x/profiles/simulation/genesis.go (hunk 4 above)
Comment (dadamu) [reply x/profiles/simulation/genesis.go:18]: The random of profilesNumber will be exactly 1 so that the `simsState.Rand.Intn(profilesNumber/2)` in the following codes occurs the panic which is `invalid argument to Intn`. So I simply changed the number to reduce.

File: x/profiles/simulation/params.go [hunk 6]
```diff
@@ -18,29 +19,33 @@ func ParamChanges(r *rand.Rand) []simtypes.ParamChange {
 		simulation.NewSimParamChange(types.ModuleName, string(types.NicknameParamsKey),
 			func(r *rand.Rand) string {
 				params := RandomNicknameParams(r)
-				return fmt.Sprintf(`{"min_nickname_len":"%s","max_nickname_len":"%s"}`,
+				return fmt.Sprintf(`{"min_length":"%s","max_length":"%s"}`,
```
Comment (bragaz) [inline x/profiles/simulation/params.go:22]: are we able to distinguish the params on tests if we have more or less the same names for both nickname and dtag params?

Comment (RiccardoM) [reply x/profiles/simulation/params.go:22]: @bragaz Yes, because they are associated to a specific key. In this case at line 19 we have `string(types.NicknameParamsKey)`

File: app/desmos/cmd/root.go (hunk 1 above)
Comment (RiccardoM) [reply app/desmos/cmd/root.go:129]: I reverted this to be `0stake` since we do not set any custom denomination anywhere

**Comment by RiccardoM** [comment]
I would like to ask both of you @bragaz @dadamu to approve this PR (I can't since I'm the one that made it) to make sure we all agree on the changed before merging it. Please do this after a detailed review

File: Makefile [hunk 7]
```diff
@@ -313,12 +313,12 @@ proto-lint:
 proto-check-breaking:
 	@$(DOCKER_BUF) breaking --against $(HTTPS_GIT)#branch=master
 
-TM_URL           = https://raw.githubusercontent.com/tendermint/tendermint/v0.34.11/proto/tendermint
+TM_URL           = https://raw.githubusercontent.com/tendermint/tendermint/v0.34.12/proto/tendermint
 GOGO_PROTO_URL   = https://raw.githubusercontent.com/regen-network/protobuf/cosmos
-COSMOS_URL 		 = https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.42.6/proto/cosmos
+COSMOS_URL 		 = https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.43.0/proto/cosmos
```
Comment (dadamu) [inline Makefile:318]:
```suggestion
COSMOS_URL 		 = https://raw.githubusercontent.com/cosmos/cosmos-sdk/v0.44.0/proto/cosmos
```

File: .changeset/config.yaml [hunk 8]
```diff
@@ -1,25 +1,58 @@
 github_repo: github.com/desmos-labs/desmos
-version: 0.17.0
+version: 1.0.2
```
Comment (bragaz) [inline .changeset/config.yaml:2]:
```suggestion
version: 1.0.3
```

File: app/desmos/cmd/root.go (hunk 1 above)
Comment (bragaz) [reply app/desmos/cmd/root.go:129]: shouldn't this become a problem when put this into mainnet one day?

File: Makefile (hunk 7 above)
Comment (RiccardoM) [reply Makefile]: Done

File: app/desmos/cmd/root.go (hunk 1 above)
Comment (RiccardoM) [reply app/desmos/cmd/root.go:129]: Ideally not. This is a simple configuration like `min_gas_prices` that is only stored locally for each validator. They will be able to update it accordingly to our instructions and then restart their node without any issue

**Review by bragaz — APPROVED** [review]
Looks ready, I think we can merge this.

File: go.mod [hunk 9]
```diff
@@ -19,19 +20,19 @@ require (
...
-	google.golang.org/grpc v1.37.0
+	google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c
+	google.golang.org/grpc v1.38.0
 	google.golang.org/protobuf v1.26.0
 )
 
 replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
 
-replace github.com/cosmos/cosmos-sdk => github.com/desmos-labs/cosmos-sdk v0.42.5-0.20210826081936-864fc99bab62
+replace github.com/cosmos/cosmos-sdk => github.com/desmos-labs/cosmos-sdk v0.43.0-alpha1.0.20210906075223-32780842edff
```
Comment (dadamu) [inline go.mod:32]: Why is cosmos-sdk replaced into v0.43.0-alpha here? I guess it should be removed.

Comment (RiccardoM) [reply go.mod:32]: @dadamu This is actually our custom version of the SDK, which integrates a particular fix I had to do to make our tests pass; https://github.com/desmos-labs/cosmos-sdk/commit/32780842edff6248ca5e49f3717ce81af56f205e

**Review by dadamu — APPROVED** [review]
Everything looks ready to be merged.
